src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MaxNumber.cs

```diff
@@ -86,93 +86,58 @@ public static void MaxNumber<T>(ReadOnlySpan<T> x, T y, Span<T> destination)
-                    // handle sNaN (it converts it to qNaN as per the now deprecated
-                    // maxNum function defined by IEEE 754:2008, but which is not inline
-                    // with the maximumNumber function that replaces it in IEEE 754:2019)
-
-                    Vector128<T> max;
-
-                    if (Sse.IsSupported && typeof(T) == typeof(float))
-                    {
-                        max = Sse.Max(x.AsSingle(), y.AsSingle()).As<float, T>();
-                    }
-                    else if (Sse2.IsSupported && typeof(T) == typeof(double))
-                    {
-                        max = Sse2.Max(x.AsDouble(), y.AsDouble()).As<double, T>();
-                    }
-                    else
-                    {
-                        max = Vector128.ConditionalSelect(Vector128.LessThan(y, x), x, y);
-                    }
-
-                    return
-                        Vector128.ConditionalSelect(Vector128.Equals(x, y),
-                            Vector128.ConditionalSelect(IsNegative(y), x, y),
-                            Vector128.ConditionalSelect(Vector128.Equals(y, y), max, x));
+                    return Vector128.ConditionalSelect(
+                        Vector128.LessThan(y, x) | IsNaN(y) | (Vector128.Equals(x, y) & IsNegative(y)),
+                        x,
+                        y
+                    );
                 }
 
                 return Vector128.Max(x, y);
+#endif
             }
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public static Vector256<T> Invoke(Vector256<T> x, Vector256<T> y)
             {
-                if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
+#if NET9_0_OR_GREATER
+                return Vector256.MaxNumber(x, y);
+#else
+                if ((typeof(T) == typeof(float)) || (typeof(T) == typeof(double)))
                 {
-                    Vector256<T> max;
-
-                    if (Avx.IsSupported && typeof(T) == typeof(float))
-                    {
-                        max = Avx.Max(x.AsSingle(), y.AsSingle()).As<float, T>();
-                    }
-                    else if (Avx.IsSupported && typeof(T) == typeof(double))
-                    {
-                        max = Avx.Max(x.AsDouble(), y.AsDouble()).As<double, T>();
-                    }
-                    else
-                    {
-                        max = Vector256.ConditionalSelect(Vector256.LessThan(y, x), x, y);
-                    }
-
-                    return
-                        Vector256.ConditionalSelect(Vector256.Equals(x, y),
-                            Vector256.ConditionalSelect(IsNegative(y), x, y),
-                            Vector256.ConditionalSelect(Vector256.Equals(y, y), max, x));
+                    return Vector256.ConditionalSelect(
+                        Vector256.LessThan(y, x) | IsNaN(y) | (Vector256.Equals(x, y) & IsNegative(y)),
```

Comment:
We don't want or need the Avx special-casing anymore?

Reply:
The Avx special casing wasn't actually any faster, the cost of `Max` and `LessThan` is the same on x64, but since we need multiple masks as the input condition, the latter lets us write a more compact algorithm that can generate some faster code, as we can combine the masks together and do a single conditional select.